go/controller/cluster.go

```diff
@@ -136,65 +77,41 @@ func (c *Cluster) SyncResource() (res autoscaler.ClusterResource, err error) {
 	if err != nil {
 		return autoscaler.ClusterResource{}, err
 	}
-
-	allReqs := make(map[v1.ResourceName]resource.Quantity)
-	allLimits := make(map[v1.ResourceName]resource.Quantity)
 	allocatable := make(v1.ResourceList)
 	for _, node := range nodeList.Items {
-		a := node.Status.Capacity
-		if len(node.Status.Allocatable) > 0 {
-			a = node.Status.Allocatable
-		}
-
-		for key, item := range a {
-			a := allocatable[key]
-			a.Add(item)
-			allocatable[key] = a
-		}
-
-		namespace := "" // get pods from all namespaces.
-		var fieldSelector fields.Selector
-		fieldSelector, err := fields.ParseSelector("spec.nodeName=" + node.Name + ",status.phase!=" + string(api.PodSucceeded) + ",status.phase!=" + string(api.PodFailed))
-		if err != nil {
-			return autoscaler.ClusterResource{}, err
-		}
-		nodeNonTerminatedPodsList, err := c.clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{FieldSelector: fieldSelector.String()})
-		if err != nil {
-			return autoscaler.ClusterResource{}, err
-		}
-
-		reqs, limits, err := getPodsTotalRequestsAndLimits(nodeNonTerminatedPodsList)
-		if err != nil {
-			return autoscaler.ClusterResource{}, err
-		}
-
-		for key, item := range reqs {
-			a := allReqs[key]
-			a.Add(item)
-			allReqs[key] = a
-		}
+		AddResourceList(&allocatable, node.Status.Allocatable)
+	}
 
-		for key, item := range limits {
-			a := allLimits[key]
-			a.Add(item)
-			allLimits[key] = a
-		}
+	// get non-terminated pods from all namespaces all nodes.
+	// FIXME(typhoonzero): scan all pods is not a efficient way.
```

Comment:
We need to get pod resource requests from all the nodes. I simplified this by get all the pods in one API call.

Reply:
If there's too much pods, this may take a lot of memory though.

Reply:
I think the memory limit is fine, 1MB can fit thousands of "medium-sized" data structures.